diffrax/_solver/foster_langevin_srk.py

```diff
@@ -0,0 +1,410 @@
+            assert transposed.shape[0] == 6
+            return jnp.polyval(transposed, h, unroll=6)
+
+        return jtu.tree_map(eval_taylor_fun, tay_coeffs)
+
+    def _recompute_coeffs(
+        self,
+        h: RealScalarLike,
+        gamma: UnderdampedLangevinX,
+        tay_coeffs: PyTree[_Coeffs],
+    ) -> _Coeffs:
+        r"""When h changes, the SRK coefficients (which depend on h) are recomputed
+        using this function."""
+        # Inner will record the tree structure of the coefficients
+        inner = sentinel = object()
+
+        def recompute_coeffs_leaf(c: UnderdampedLangevinLeaf, _tay_coeffs: _Coeffs):
+            # c is a leaf of gamma
+            # Depending on the size of h*gamma choose whether the Taylor expansion or
+            # direct computation is more accurate.
+            cond = h * c < self.taylor_threshold
+            tay_out = self._eval_taylor(h, _tay_coeffs)
+            if cond.ndim < jtu.tree_leaves(tay_out)[0].ndim:
+                # This happens when c is a scalar
+                cond = jnp.expand_dims(cond, axis=-1)
+
+            def select_tay_or_direct():
+                if jnp.ndim(c) == 0:
+                    direct_out = self._directly_compute_coeffs_leaf(h, c)
+                else:
+                    fun = lambda _c: self._directly_compute_coeffs_leaf(h, _c)
+                    direct_out = vmap(fun)(c)
+
+                def _choose(tay_leaf, direct_leaf):
+                    assert tay_leaf.ndim == direct_leaf.ndim == cond.ndim, (
+                        f"tay_leaf.ndim: {tay_leaf.ndim},"
+                        f" direct_leaf.ndim: {direct_leaf.ndim},"
+                        f" cond.ndim: {cond.ndim}"
+                    )
+                    return jnp.where(cond, tay_leaf, direct_leaf)
+
+                return jtu.tree_map(_choose, tay_out, direct_out)
+
+            # If all entries of h*gamma are below threshold, only compute tay_out
+            # otherwise, compute both tay_out and direct_out and select the
+            # correct one for each dimension
+            out = lax.cond(
+                eqxi.unvmap_all(cond),
+                lambda: tay_out,
+                select_tay_or_direct,
+            )
+
+            # The inner tree structure is just the structure of _Coeffs,
+            # but we need to record it for the tree transpose
+            nonlocal inner
+            if inner is sentinel:
+                inner = jtu.tree_structure(out)
+            else:
+                assert (
+                    jtu.tree_structure(out) == inner
+                ), f"Expected {inner}, got {jtu.tree_structure(out)}"
+
+            return out
+
+        tree_with_coeffs = jtu.tree_map(recompute_coeffs_leaf, gamma, tay_coeffs)
+        outer = jtu.tree_structure(gamma)
+        assert inner is not sentinel, "inner tree structure not set"
+        coeffs_with_tree = jtu.tree_transpose(outer, inner, tree_with_coeffs)  # type: ignore
```

Comment:
nit: can you make this `# pyright: ignore[someErrorCode]`?

We only use pyright so I prefer to use the more specific annotation.